src/plugins/data/public/query/query_string/language_service/get_query_control_links.tsx

```diff
@@ -0,0 +1,172 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { i18n } from '@osd/i18n';
+import React, { useState } from 'react';
+import {
+  EuiButtonIcon,
+  EuiFlexGroup,
+  EuiFlexItem,
+  EuiLink,
+  EuiPopoverTitle,
+  EuiText,
+  EuiWrappingPopover,
+} from '@elastic/eui';
+import ReactDOM from 'react-dom';
+import { FormattedMessage } from 'react-intl';
+import {
+  OpenSearchDashboardsContextProvider,
+  toMountPoint,
+} from '../../../../../opensearch_dashboards_react/public';
+import { IDataPluginServices } from '../../../types';
+import { PPLReferenceFlyout } from './ppl_reference_flyout';
+
+export interface QueryControl {
+  id: string;
+  label: string;
+  testId: string;
+  ariaLabel: string;
+  run: (anchorElement: HTMLElement) => void;
+  iconType: string;
+}
+
+export const QueryControls = (props: {
+  services: IDataPluginServices;
+  queryLanguage: string;
+  onToggleCollapse: () => void;
+  savedQueryManagement?: any;
+  additionalControls?: QueryControl[];
+}) => {
+  const [isCollapsed, setIsCollapsed] = useState<boolean>(false);
+  const [isLanguageReferenceOpen, setIsLanguageReferenceOpen] = useState<boolean>(false);
+
+  const languageReferenceContainer = document.createElement('div');
+
+  const onCloseLanguageReference = () => {
+    ReactDOM.unmountComponentAtNode(languageReferenceContainer);
+    setIsLanguageReferenceOpen(false);
+  };
+
+  const osdDQLDocs = 'https://opensearch.org/docs/2.16/dashboards/dql)';
+  const dqlFullName = (
+    <FormattedMessage
+      id="data.query.queryBar.dqlFullLanguageName"
+      defaultMessage="OpenSearch Dashboards Query Language"
+    />
+  );
+
+  const languageReference: QueryControl = {
+    id: 'languageReference',
+    label: i18n.translate('discover.queryControls.languageReference', {
+      defaultMessage: 'Open',
+    }),
+    testId: 'languageReference',
+    ariaLabel: i18n.translate('discover.queryControls.languageReference', {
+      defaultMessage: `Language Reference`,
+    }),
+    run: async (anchorElement) => {
+      if (props.queryLanguage === 'PPL' || props.queryLanguage === 'SQL') {
```

Comment:
To keep the code cleaner, cant we register these when we register the PPL and SQL language using the additional controls property?